### ISSUE TITLE
Railway platforms with covered=yes should be rendered again

### ISSUE THREAD

**Issue by MaartenDeen** [gravitystorm/openstreetmap-carto#3421]
I came across this railway platform that does not get rendered anymore: https://www.openstreetmap.org/way/288846041
An adjacent platform is getting rendered: https://www.openstreetmap.org/way/548528350

The only difference I see is covered=yes on the one that does not get rendered. I'm sure it was being rendered in earlier versions of the stylesheet.

**Comment by Tomasz-W** [comment]
Yes, it's because of covered=yes tag. See https://github.com/gravitystorm/openstreetmap-carto/pull/3162#issuecomment-397911519

**Comment by kocio-pl** [comment]
Thanks for the report.

This helps to avoid situations like this:

![dzwut9gf](https://user-images.githubusercontent.com/5439713/39503259-43d7179e-4dc5-11e8-98ed-631a6798bfb1.png)

If the tagging is more specific (like `covered=trees` or `covered=partial`), platforms are being rendered.

**Comment by matthijsmelissen** [comment]
I don't see a reason not to render covered platforms. It might just mean that the platform has a roof over its full length. Not rendering underground platforms makes more sense, but covered=yes does not imply underground.

**Comment by matthijsmelissen** [comment]
I don't see a reason not to render covered platforms. It might just mean that the platform has a roof over its full length. Not rendering underground platforms makes more sense, but covered=yes does not imply underground.

**Comment by kocio-pl** [comment]
There was a report on this before - what do you think about this reply? https://github.com/gravitystorm/openstreetmap-carto/pull/3162#issuecomment-396536436

**Comment by kocio-pl** [comment]
> It might just mean that the platform has a roof over its full length.

I was assuming exactly this and found it to be a problem too.

**Comment by MaartenDeen** [comment]
That issue is about clearly defined underground platforms, tagged with location=underground. This platform does not have such a tag so it should not be treated as such.
In addition, then why is this way rendered? https://www.openstreetmap.org/way/288846045 
It has the exact same tags as https://www.openstreetmap.org/way/288846041 which is not rendered.

**Comment by kocio-pl** [comment]
I would not go with trying to explain it in the framework of underground, as I said I was aware that this is another, but similar kind of problem (without assumption that it is underground). So please let's focus on the question whether covered should or should not be rendered (and why yes or why not).

Tagging on this station is not clear to me - there's also https://www.openstreetmap.org/way/288846038, which looks that platform 2 is probably duplicated and some of them overlap. Please review this and fix if possible.

**Comment by kocio-pl** [comment]
Quick investigation (exporting OSM file and replacing names with something like 2x, 2y etc): there are like 4 objects named "Perron 2", and the rendered name comes from https://www.openstreetmap.org/relation/7831438, because it has no `covered=yes` tagging. 

Are you sure you need all 4 with the same name "Perron 2"?

**Comment by naomap** [comment]
Many railway platforms are not rendered indeed, for instance https://www.openstreetmap.org/way/140249174 which holds the tag covered=yes. This is indeed different from location=underground and I think it should be rendered.

This seems to be caused by this piece of code, line 688 in project.mml
 AND (covered NOT IN ('yes') OR covered IS NULL))
That doesn't mean I feel able to fix it !

**Comment by jragusa** [comment]
I think the tag `cover` is not appropriate for such situation. They are not _covered_ by a superstructure like a glass vault (verrière) such as at the Bordeaux train station. `shelter` is much more appropriate for a marquise, especially because the platforms are not entirely _covered_.

**Comment by aceman444** [comment]
I also wondered why https://www.openstreetmap.org/way/388148758 is not rendered and came across this issue.

Notice a close highway=platform (a bus platform) is rendered fine even when covered=yes: https://www.openstreetmap.org/way/126362312 .

**Comment by imagico** [comment]
This seems to have been fixed by #4797.
